### PR TITLE
Use `InvalidMessageException` and add test

### PR DESCRIPTION
**Pull request by ijuma**

### CODE REVIEW

**Comment by gwenshap** [comment]
LGTM. 
Love the new low-level test methods and love having negative tests.

File: core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    assertEquals(1, produceResponse.responses.size)
+    val (tp, partitionResponse) = produceResponse.responses.asScala.head
+    assertEquals(topicPartition, tp)
+    assertEquals(Errors.NONE.code, partitionResponse.errorCode)
+    assertEquals(0, partitionResponse.baseOffset)
+    assertEquals(-1, partitionResponse.timestamp)
+  }
+
+  /* returns a pair of partition id and leader id */
+  private def createTopicAndFindPartionWithLeader(topic: String): (Int, Int) = {
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala]: Just noticed a typo in this method name. Will fix it.

Comment (ijuma) [inline core/src/test/scala/unit/kafka/server/ProduceRequestTest.scala]: Fixed.

**Comment by junrao** [comment]
The patch LGTM. It's a bit weird that Errors and ErrorMapping map error code 2 to different exception names. Perhaps we can add a comment in InvalidMessageException that when we phase out the old scala clients, we can replace InvalidMessageException with CorruptRecordException.

**Comment by ijuma** [comment]
@junrao We kind of say that already:

``` scala
* InvalidMessageException extends CorruptRecordException for temporary compatibility with the old Scala clients.
```

**Comment by ijuma** [comment]
But perhaps that could be made clearer.

**Pull request merged**